include/musica/micm.hpp

```diff
@@ -162,20 +202,126 @@ namespace musica
     template<class T>
     T GetSpeciesProperty(const std::string &species_name, const std::string &property_name, Error *error);
 
-    static constexpr std::size_t NUM_GRID_CELLS = 1;
+   public:
+    short solver_type_;
 
-   private:
-    using DenseMatrixPolicy = micm::Matrix<double>;
-    using SparseMatrixPolicy = micm::SparseMatrix<double, micm::SparseMatrixStandardOrdering>;
-    using SolverPolicy = typename micm::RosenbrockSolverParameters::
-        template SolverType<micm::ProcessSet, micm::LinearSolver<SparseMatrixPolicy, micm::LuDecomposition>>;
-    using Rosenbrock = micm::Solver<SolverPolicy, micm::State<DenseMatrixPolicy, SparseMatrixPolicy>>;
+    /// @brief Vector-ordered Rosenbrock solver type
+    using DenseMatrixVector = micm::VectorMatrix<double, MICM_VECTOR_MATRIX_SIZE>;
+    using SparseMatrixVector = micm::SparseMatrix<double, micm::SparseMatrixVectorOrdering<MICM_VECTOR_MATRIX_SIZE>>;
+    using RosenbrockVectorType = typename micm::RosenbrockSolverParameters::
+        template SolverType<micm::ProcessSet, micm::LinearSolver<SparseMatrixVector, micm::LuDecomposition>>;
+    using Rosenbrock = micm::Solver<RosenbrockVectorType, micm::State<DenseMatrixVector, SparseMatrixVector>>;
+    std::unique_ptr<Rosenbrock> rosenbrock_;
 
-    std::unique_ptr<Rosenbrock> solver_;
+    /// @brief Standard-ordered Rosenbrock solver type
+    using DenseMatrixStandard = micm::Matrix<double>;
+    using SparseMatrixStandard = micm::SparseMatrix<double, micm::SparseMatrixStandardOrdering>;
+    using RosenbrockStandardType = typename micm::RosenbrockSolverParameters::
+        template SolverType<micm::ProcessSet, micm::LinearSolver<SparseMatrixStandard, micm::LuDecomposition>>;
+    using RosenbrockStandard = micm::Solver<RosenbrockStandardType, micm::State<DenseMatrixStandard, SparseMatrixStandard>>;
+    std::unique_ptr<RosenbrockStandard> rosenbrock_standard_;
 
+   private:
     std::unique_ptr<micm::SolverParameters> solver_parameters_;
   };
 
+  template<class T>
+  inline void MICM::Solve(
+      std::unique_ptr<T> &solver,
```

Comment:
could this just be:
```
auto &solver
```
to avoid having a template parameter?

Reply:
Yes, done!